core/src/test/java/org/apache/iceberg/TestPaths.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import java.net.URISyntaxException;
+import org.apache.hadoop.fs.Path;
+import org.junit.Test;
+
+public class TestPaths {
```

Comment:
I think it would be better to add a test to `TestReachableFileUtil` than adding a new test class

Reply:
@nastra this happens only with S3 file paths and not with hadoop file paths. Unfortunately, all our tests use local paths using hadoop Path objects. I could not find integration tests with S3 paths but using hadoop fileIO. Do you have a suggestion on how this can be tested?